src/els_indexer.erl

```diff
@@ -208,23 +209,21 @@ index_document(Uri, Text, sync) ->
   ok = index(Uri, Text).
 
 -spec register_reference(uri(), poi()) -> ok.
-register_reference( Uri
-                  , #{kind := record_expr, id := Name, range := Range}
-                  ) ->
-  els_dt_references:insert(#{ id    => {record, Name}
+register_reference(Uri, #{kind := Kind, id := RecordName, range := Range})
+  when Kind =:= record_expr;
+       Kind =:= record_access ->
+  els_dt_references:insert(#{ id    => {record, RecordName}
                             , uri   => Uri
                             , range => Range
                             });
-register_reference( Uri
-                  , #{kind := record_access, id := {Name, _}, range := Range}
-                  ) ->
-  els_dt_references:insert(#{ id    => {record, Name}
+register_reference(Uri, #{kind := macro, id := MacroName, range := Range}) ->
+  els_dt_references:insert(#{ id    => {macro, MacroName}
                             , uri   => Uri
                             , range => Range
                             });
 register_reference(Uri, #{id := {F, A}} = POI) ->
```

Comment:
Somewhat unrelated, but shouldn't we match on the `kind` here as well?

Reply:
Yup, but we weren't doing it and I had already introduced other unrelated changes.